src/snowflake/cli/api/sql_execution.py

```diff
@@ -21,11 +21,11 @@
 from textwrap import dedent
 from typing import Iterable, Optional, Tuple
 
+from snowflake.cli._plugins.nativeapp.sf_facade_constants import UseObjectType
+from snowflake.cli._plugins.nativeapp.sf_facade_exceptions import CouldNotUseObjectError
```

Comment:
Unsure if we want to move these into the `api` folder if we're going this route or leave it like this

Reply:
the facede is nativeapp-only for now, we'd move the whole thing later if we wanted to

Reply:
we probably shouldn't be importing from plugins from core code though, does this file need to change?

Reply:
If we're using the sql facade version of the exception that has the proper exception hierarchy that we want then we would need to import it from `snowflake.cli._plugins.nativeapp.sf_facade_exceptions`, unless we want to use a different error

Reply:
i was under the impression that we wanted to keep developing the SQL facade stuff separate from the rest of the CLI so we don't have to constantly get approval from them. i'd vote for just leaving any non-nativeapp file as-is. If that's not doable (or it would interfere with our metrics), we should move the enum and error class to the core CLI so that core files don't import from plugins (at some point we had a pre-commit check for it, not sure how this passed in the first place)

Reply:
yeah, there are way too many changes here. Can we take a step back and see what's required? Our strategy so far is that the facade is NA-specific, since it's under active development. We're not changing the rest of the CLI, that introduces more risk that we're not ready for ATM.